docarray/typing/tensor/tensorflow_tensor.py

```diff
@@ -0,0 +1,224 @@
+from typing import TYPE_CHECKING, Any, Dict, Generic, Type, TypeVar, Union, cast
+
+import numpy as np
+import tensorflow as tf  # type: ignore
+
+from docarray.typing.proto_register import _register_proto
+from docarray.typing.tensor.abstract_tensor import AbstractTensor
+
+if TYPE_CHECKING:
+    from pydantic.fields import ModelField
+    from pydantic import BaseConfig
+    from docarray.proto import NdArrayProto
+    from docarray.computation.tensorflow_backend import TensorFlowCompBackend
+
+from docarray.base_document.base_node import BaseNode
+
+T = TypeVar('T', bound='TensorFlowTensor')
+ShapeT = TypeVar('ShapeT')
+
+tf_base: type = type(tf.Tensor)
+node_base: type = type(BaseNode)
+
+
+# the mypy error suppression below should not be necessary anymore once the following
+# is released in mypy: https://github.com/python/mypy/pull/14135
+class metaTensorFlow(
+    AbstractTensor.__parametrized_meta__,  # type: ignore
+    node_base,  # type: ignore
+    tf_base,  # type: ignore
+):  # type: ignore
+    pass
+
+
+@_register_proto(proto_type_name='tensorflow_tensor')
+class TensorFlowTensor(AbstractTensor, Generic[ShapeT], metaclass=metaTensorFlow):
```

Comment:
I think you need to explain that contratry to torch and ndarray we cannot use the TensorFLowTensor as if it was a tf.tensor because we cannot subclass and therefore user have to use unwrap() or .tensor directly